### ISSUE TITLE
HA DNS issues when using local resolver / unwanted failover to cloudflare dns

### ISSUE THREAD

**Issue by jspanitz** [home-assistant/supervisor#2966]
### Describe the issue you are experiencing

When HA is behind a firewall and the firewall also hosts DNS services, HA fails to perform DNS queries successfully.  This is true when the only configured DNS is the local DNS.  This is especially apparent when firewall blocks all outbound DNS.  HA becomes sluggish.

### What is the used version of the Supervisor?

supervisor-2021.06.3

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

Home Assistant OS 6.0

### What version of Home Assistant Core is installed?

core-2021.6.4

### Steps to reproduce the issue

Configure HA network settings in GUi to use local firewall for DNS.
Ensure no other DNS services are configured by issuing "dns reset" from HA cli.
Verify DNS by issuing "dns info" 

![image](https://user-images.githubusercontent.com/51183445/122497648-bbb57f00-cfbb-11eb-81d7-23671f9ab330.png)

Firewall logs still show HA reaching out to 1.1.1.1 and 1.0.0.1 on tcp/53 and tcp/853 even though cloudflare dns is not configured anywhere on the system.

System exhibits slow and sluggish behavior due to the unpermitted cloudflare dns traffic.

### Anything in the Supervisor logs that might be useful for us?

```txt
21-06-18 02:20:23 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository
21-06-18 02:20:28 ERROR (MainThread) [supervisor.store.git] Can't update https://github.com/hassio-addons/repository repo: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v --update-shallow --depth=1 origin
  stderr: 'fatal: unable to access 'https://github.com/hassio-addons/repository/': Could not resolve host: github.com'.
21-06-18 02:20:28 INFO (MainThread) [supervisor.resolution.module] Create new issue IssueType.CORRUPT_REPOSITORY - ContextType.STORE / a0d7b954
21-06-18 02:20:28 INFO (MainThread) [supervisor.resolution.module] Create new suggestion SuggestionType.EXECUTE_RESET - ContextType.STORE / a0d7b954
21-06-18 02:20:28 ERROR (MainThread) [supervisor.store.git] Can't update https://github.com/home-assistant/addons repo: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v --update-shallow --depth=1 origin
  stderr: 'fatal: unable to access 'https://github.com/home-assistant/addons/': Could not resolve host: github.com'.
21-06-18 02:20:28 INFO (MainThread) [supervisor.resolution.module] Create new issue IssueType.CORRUPT_REPOSITORY - ContextType.STORE / core
21-06-18 02:20:28 INFO (MainThread) [supervisor.resolution.module] Create new suggestion SuggestionType.EXECUTE_RESET - ContextType.STORE / core
21-06-18 02:20:28 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-2367' coro=<Repository.update() done, defined at /usr/src/supervisor/supervisor/store/repository.py:106> exception=StoreGitError()>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/git.py", line 155, in pull
    await self.sys_run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 828, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 702, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "/usr/local/lib/python3.9/site-packages/git/cmd.py", line 450, in wait
    raise GitCommandError(remove_password_if_present(self.args), status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v --update-shallow --depth=1 origin
  stderr: 'fatal: unable to access 'https://github.com/home-assistant/addons/': Could not resolve host: github.com'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/repository.py", line 110, in update
    await self.git.pull()
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 107, in wrapper
    raise err
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 105, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/store/git.py", line 195, in pull
    raise StoreGitError() from err
supervisor.exceptions.StoreGitError
21-06-18 02:20:28 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-2366' coro=<Repository.update() done, defined at /usr/src/supervisor/supervisor/store/repository.py:106> exception=StoreGitError()>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/git.py", line 155, in pull
    await self.sys_run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 828, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 702, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "/usr/local/lib/python3.9/site-packages/git/cmd.py", line 450, in wait
    raise GitCommandError(remove_password_if_present(self.args), status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v --update-shallow --depth=1 origin
  stderr: 'fatal: unable to access 'https://github.com/hassio-addons/repository/': Could not resolve host: github.com'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/repository.py", line 110, in update
    await self.git.pull()
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 107, in wrapper
    raise err
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 105, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/store/git.py", line 195, in pull
    raise StoreGitError() from err
supervisor.exceptions.StoreGitError
21-06-18 02:20:29 INFO (MainThread) [supervisor.store] Loading add-ons from store: 63 all - 0 new - 0 remove
```

### Additional information

HA DNS appears to have issues with local resolvers.

**Comment by pvizeli** [comment]
https://github.com/home-assistant/plugin-dns/issues/22

**Comment by jspanitz** [comment]
@pvizeli The other thread was closed on January 30th and raised a privacy issue (which by the way I feel is very valid as well) which is not the issue I raised here. 

The issue here is that if outbound DNS is blocked and you force HA to use you internal DNS, it becomes sluggish and the add-ons are no longer accessible.   The odd part is HA OS updates still seem to show up in the GUI and can be installed, which would indicate that DNS is working.   A 3rd party add on, HACS, also continues to show updates and it's repo is visible.

There is clearly an HA DNS issue that has been reported and discusses by many, many users.  I'd like to help the team get it resolved.  If there is no interest in doing so, I'd request the team state as much rather than just closing tickets with no explanation.
